plugin/upstream_generator/evaluator/yamlpath_evaluator.go

```diff
@@ -0,0 +1,69 @@
+package evaluator
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/goto/salt/log"
+	"github.com/vmware-labs/yaml-jsonpath/pkg/yamlpath"
+	"gopkg.in/yaml.v3"
+
+	"github.com/goto/optimus/internal/errors"
+)
+
+type yamlpathEvaluator struct {
```

Comment:
why don't we export it? asking because the implementation under `newYamlpathEvaluator` treat as if it's exposed.

Reply:
simply because we have to create this object from evaluator factory, there's no need to expose this object

Reply:
sure, I'm just saying that we treat `newYamlpathEvaluator` as if it is exported, in away we have some validations there.